File: 3-Tutorial/Praktikum 3/1/RekeningGiro.cpp
// Nama 	: Naufal Dean Anugrah
// NIM		: 13518123
// Tanggal	: 6 Februari 2020
// Topik 	: Inheritance

#include <iostream>
#include "RekeningGiro.h"

using namespace std;

// Konstruktor menginisialisi saldo (parameter 1) dan suku bunga (parameter 2)
// Set suku bunga 0.0 apabila suku bunga bernilai negatif
RekeningGiro::RekeningGiro(double saldo, double sukuBunga): Rekening(saldo) {
	this->sukuBunga = (sukuBunga < 0.0) ? (0.0) : (sukuBunga);
}

// Getter, Setter
void RekeningGiro::setSukuBunga(double sukuBunga) {
	this->sukuBunga = sukuBunga;
}
double RekeningGiro::getSukuBunga() const {
	return this->sukuBunga;
}

// Member Function
// Bunga dihitung dari saldo dikali suku bunga
double RekeningGiro::hitungBunga() {
	return Rekening::getSaldo() * this->sukuBunga;
}

File: Java/7-Tutorial/3/RekeningGiro.java
// Nama		: Naufal Dean Anugrah
// NIM		: 13518123
// Tanggal	: 12 Maret 2020
// Topik 	: Pengenalan java

class RekeningGiro extends Rekening {
	private boolean penalti;

	public RekeningGiro(String nama, double saldo) {
		super(nama, saldo, 0.07);
		if (this.saldo < 500)
			this.penalti = true;
		else
			this.penalti = false;
	}

	public void setor(double uang) {
		this.saldo += uang - 0.10;
		if (this.saldo < 500)
			this.penalti = true;
	}

	public void tarik(double uang) {
		if (uang + 0.10 <= this.saldo) {
			this.saldo -= uang + 0.10;
		}
		if (this.saldo < 500)
			this.penalti = true;
	}

	public void update() {
		this.saldo += this.getSukuBunga() * this.saldo - this.hitungBiaya();
		if (this.penalti) {
			this.saldo -= 10;
		}
		if (this.saldo >= 500) {
			this.penalti = false;
		}
		if (this.saldo < 0) {
			this.saldo = 0;
		}
	}
}
